merkletree/proof.go

```diff
@@ -88,13 +89,22 @@ func VerifyAuthPath(ap *AuthenticationPath,
 	return true
 }
 
+func VerifyCommitment(salt []byte, key string, value []byte, commitment []byte) bool {
+	got := crypto.Digest(salt, []byte(key), value)
```

Comment:
> What do you think?

That's probably fine.

I'm wondering if `salt` is particular to our implementation.  And, instead, we should model it more like,

```
type Commit struct {
  salt []byte
  value []byte
}

func NewCommit(stuff) (*Commit, error) {
  salt, err := MakeRand()
  if err != nil {
    return nil, err
  }
  return Digest(salt, stuff), nil
}

func (c *Commmit) Verify(stuff []byte) bool {
  return bytes.Equal(c.value, Digest(c.salt, stuff)
}
```

Maybe that's overkill?  On the client you'd skip the constructor,

```
&Commitment{salt, value}
```

I don't know.  I think what you suggested is fine ...